restapi/src/main/java/io/stargate/web/docsapi/service/json/DeadLeafCollectorImpl.java

```diff
@@ -0,0 +1,37 @@
+package io.stargate.web.docsapi.service.json;
+
+import java.util.*;
+
+public class DeadLeafCollectorImpl implements DeadLeafCollector {
+  private Map<String, Set<DeadLeaf>> deadPaths;
+
+  public DeadLeafCollectorImpl() {
+    deadPaths = new HashMap<>();
+  }
+
+  public void addLeaf(String path, DeadLeaf leaf) {
+    Set<DeadLeaf> leavesAtPath = deadPaths.getOrDefault(path, new HashSet<>());
+    leavesAtPath.add(leaf);
+    deadPaths.put(path, leavesAtPath);
+  }
+
+  public void addArray(String path) {
+    Set<DeadLeaf> leavesAtPath = deadPaths.getOrDefault(path, new HashSet<>());
+    leavesAtPath.add(ImmutableDeadLeaf.builder().name(DeadLeaf.ARRAY).build());
```

Comment:
Could we just make `ImmutableDeadLeaf.builder().name(DeadLeaf.ARRAY).build()` a constant on the `DeadLeaf` interface to avoid re-creating the same object?